### ISSUE TITLE
Why was Tabular removed?

### ISSUE THREAD

**Issue by testvinder** [keras-team/autokeras#562]
Why was the Tabular module removed? Are you open to contributions on improving/bringing back the tabular module?

**Comment by jhfjhfj1** [comment]
@testvinder This has been removed because Auto-Keras is focusing on deep learning tasks.
We are moving it to a new project just focusing on tabular data, which will be completed soon.
By the time we complete the basics, we will be open to contribution.
Thank you!

**Comment by flamby** [comment]
@jhfjhfj1 I found odd that this module was based on lightgbm. 
I'm currently using keras (tensorflow backend) for timeseries binary classification (continuous data ; finance) using Conv1D, MaxPooling1D, Bidirectional, and finally LSTM or GRU. 
NAS support would be great for such DL architectures on tabular data.
Is the new project already on github, and planned to rely on lightgbm only?

**Comment by SlickJackson** [comment]
Hey,
I got the same question... right now I'm using hyperopt and hyperas to optimize any deterministic regression model... for autoKeras, it seems to be a small leap to use tabular data... where can we find the other project?

**Comment by csteel45** [comment]
Are there any updates on this? I am using a CNN in a GAN to create synthetic data sets of tabular data. Considering that 95% of the enterprise data today is in tabular format, I would think tabular data support would be in high demand.

**Comment by bolkedebruin** [comment]
I have created a pypi package (https://pypi.org/project/autokeras-tabular/) that reinstates the missing functionality. Happy to remove it when the new project is there.

Obviously it will have the same pros and cons as the original did.

**Comment by zhanwenchen** [comment]
I'm going to attempt to explain this thread. Back in January 2019, @jhfjhfj1 pushed a "Tabular"/"GeneralClassifier" API (#406) but removed it soon after in February (#550). The main source code for this "Tabular" module was [here](https://github.com/keras-team/autokeras/blob/4ddd568b06b4045ace777bc0fb7bc18573b85a75/autokeras/tabular/tabular_supervised.py), which has now been moved by @bolkedebruin to [autokeras_tabular](https://github.com/bolkedebruin/autokeras_tabular).

The author, @jhfjhfj1, explained the rationale behind the removal as such:

> @testvinder This has been removed because Auto-Keras is focusing on deep learning tasks.
> We are moving it to a new project just focusing on tabular data, which will be completed soon.
> By the time we complete the basics, we will be open to contribution.
> Thank you!

At first, it wasn't clear what @jhfjhfj1 meant by "focusing on deep learning tasks," because regressions are in fact a core deep learning task. However, upon a closer look at the removed module code, as @flamby suggests, the module used [LightGBM](https://lightgbm.readthedocs.io/en/latest/), a "gradient boosting framework that uses tree based learning algorithms." In other words, it is machine learning but not deep learning.

It is still not clear to me why @jhfjhfj1 did not implement this "Tabular"/"GeneralClassifier" with deep learning like a fully-connected net or any sort of CNN/RNN like the "image" and "text" modules. Why use machine learning in *neural* architecture search? Besides, NAS/AutoML as a paradigm should be domain- and problem-independent; otherwise, aren't we simply resorting back to manual search with bells and whistles ("search heuristics")?
